app/actions/handlers.py

```diff
@@ -47,6 +46,33 @@ def extract_gmt_offsets(transmissions, integration_id):
     return offsets_by_device
 
 
+def get_file_group_by_status(status):
+    if status == FileStatus.PENDING:
+        return PENDING_FILES
+    elif status == FileStatus.IN_PROGRESS:
+        return IN_PROGRESS_FILES
+    elif status == FileStatus.PROCESSED:
+        return PROCESSED_FILES
+    else:
+        return None
+
+
+async def get_file_status(file_name):
+    is_a_pending_file = await state_manager.group_ismember(PENDING_FILES, file_name)
+    if is_a_pending_file:
+        return {"status": FileStatus.PENDING.value, "group": PENDING_FILES}
+    else:
+        is_an_in_progress_file = await state_manager.group_ismember(IN_PROGRESS_FILES, file_name)
+        if is_an_in_progress_file:
+            return {"status": FileStatus.IN_PROGRESS.value, "group": IN_PROGRESS_FILES}
+        else:
+            is_a_processed_file = await state_manager.group_ismember(PROCESSED_FILES, file_name)
+            if is_a_processed_file:
+                return {"status": FileStatus.PROCESSED.value, "group": PROCESSED_FILES}
+            else:
+                return {}
```

Comment:
Just a suggestion to simplify the code a little bit:
```suggestion
    if await state_manager.group_ismember(PENDING_FILES, file_name):
        return {"status": FileStatus.PENDING.value, "group": PENDING_FILES}
    elif await state_manager.group_ismember(IN_PROGRESS_FILES, file_name):
        return {"status": FileStatus.IN_PROGRESS.value, "group": IN_PROGRESS_FILES}
    elif await state_manager.group_ismember(PROCESSED_FILES, file_name):
        return {"status": FileStatus.PROCESSED.value, "group": PROCESSED_FILES}
    return {}
```